beangulp/importers/csv.py

```diff
@@ -327,14 +333,23 @@ def get(row, ftype):
 
         # Add a balance entry if possible
         if Col.BALANCE in iconfig and entries:
-            entry = entries[-1]
+            entry = None
+            if Col.CURRENCY in iconfig:
+                for e in entries[::-1]:
+                    balance = e.meta.get('balance', None)
+                    if balance and balance[1] == self.currency:
+                        entry = e
+                        break
+
+            if not entry:
+                entry = entries[-1]
```

Comment:
I have the impression that this does not do what intended if there are no transactions in the main currency in the CSV file. Also, wouldn't it be better to emit one `balance` directive for each currency in the file?

Reply:
Yes I was also thinking about producing a balance per currency however I stopped at one because in my specific case I did not need it to do so. If you think it's required I can put the functionality in.